lxd/operations/operations.go

```diff
@@ -107,21 +109,45 @@ type Operation struct {
 	onConnect func(*Operation, *http.Request, http.ResponseWriter) error
 	onDone    func(*Operation)
 
-	// Indicates if operation has finished.
-	finished cancel.Canceller
-
 	// Locking for concurent access to the Operation
 	lock sync.Mutex
 
 	state  *state.State
 	events *events.Server
 }
 
+func (op *Operation) setupContext(inCtx context.Context, s *state.State) error {
+	opCtx := context.Background()
+	if s != nil {
+		opCtx = s.ShutdownCtx
+	}
```

Comment:
Thinking a little bit more about this. Should an operation only ever have a `context.Background()` plus any extra contents (such as the requestor)?

If the daemon is shut down, then the shutdown context will cancel, and all operations will cancel. This would mean that LXD wouldn't wait for running exec websockets to close before shutting down.

I'll experiment with this

Reply:
Hmm ok.. Well the operation still has access to the shutdown context via the state, so the particular operation can choose to act on that independently of its own context. I will change this commit so that all operations have a background context plus any extra "contextual" information :)

Reply:
Fine for now to limit the scope of the change.

Reply:
Done. Will unmark as draft if I have a clean test run